packages/gatsby/src/query/index.js

```diff
@@ -305,7 +307,10 @@ const startListeningToDevelopQueue = ({ graphqlTracing } = {}) => {
   listenerQueue = new Queue((queryJobs, callback) => {
     const activity = createQueryRunningActivity(queryJobs.length)
 
-    const onFinish = (...arg) => {
+    const onFinish = async (...arg) => {
+      if (!webpackStatusUtil.isPending()) {
```

Comment:
Do we have a guarantee that it's pending due to pages needing a flush or that a flush will always happen?

Reply:
Flush only flushes necessary pending changes (that are in Redux) so this is fine to call always. 

Guaranteed to always run.

Reply:
https://github.com/gatsbyjs/gatsby/pull/24808#discussion_r435936784